### PR TITLE
Fix migrations & initializations when using multiple workers

### PR DESCRIPTION
**Pull request by Petitoto**
### Description

This PR fixes migrations when working with multiple workers.

It allows to run migrations and initializations in a synchronous way before lifespan. Thus, gunicorn can preload the app to run them before creating workers.

### Checklist

- [ ] Created tests which fail without the change (if possible)
- [x] All tests passing
- [ ] Extended the documentation, if necessary

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -12,5 +12,6 @@ unidecode==1.3.7
 uvicorn[standard]==0.23.2
 redis==5.0.1
 icalendar == 5.0.11
+psycopg2==2.9.9
```
Comment (armanddidierjean) [inline requirements.txt:15]: Could you add a comment explaining what the dependency is used for?

File: app/app.py [hunk 2]
```diff
@@ -261,6 +244,26 @@ async def startup(app: FastAPI):
...
+    initialize_groups(sync_engine)
+    initialize_module_visibility(sync_engine)
+
+    # Initialize Redis
+    if (
+        app.dependency_overrides.get(get_redis_client, get_redis_client)(
+            settings=settings
+        )
+        is False
+    ):
```
Comment (armanddidierjean) [inline app/app.py:264]:
```suggestion
    if (
        not app.dependency_overrides.get(get_redis_client, get_redis_client)(
            settings=settings
        )
    ):
```

File: app/app.py [hunk 3]
```diff
@@ -261,6 +244,26 @@ async def startup(app: FastAPI):
         allow_headers=["*"],
     )
 
+    # Initialize database connection
+    app.dependency_overrides.get(get_db_engine, get_db_engine)(settings=settings)
```
Comment (armanddidierjean) [inline app/app.py:248]: Is this useful? We don't use the returned engine.

**Review by armanddidierjean — CHANGES_REQUESTED** [review]

File: app/app.py (hunk 3 above)
Comment (Petitoto) [reply app/app.py:248]: It creates the Async Engine, which is stored as a global variable and then used by all workers

**Pull request merged**